ada/views.py

```diff
@@ -0,0 +1,34 @@
+from urllib.parse import urlparse
+
+from django.http import Http404, HttpRequest, HttpResponse
+from django.shortcuts import HttpResponseRedirect, render
+
+from .models import RedirectUrl, RedirectUrlsEntry
+
+
+def clickActivity(request: HttpRequest, pk: int, whatsappid: int) -> HttpResponse:
+    try:
+        redirect_url = RedirectUrl.objects.get(id=pk)
+    except RedirectUrl.DoesNotExist:
+        raise Http404()
+    else:
+        store_url_entry = RedirectUrlsEntry(url=redirect_url)
+        store_url_entry.save()
+        url = f"{redirect_url.symptom_check_url}" f"?whatsappid={whatsappid}"
+        parsed_url = urlparse(url)
+        destination_url = parsed_url.geturl()
```

Comment:
This will just make sure that if it's a valid URL, then it carries on, but if it's an invalid URL, then it raises an exception and results in a 500 error. That's not quite what we want to do here. We rather want to use https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote to quote the querystring part of the URL, and then combine the URL and the querystring to create the destination URL